camel/toolkits/notion_toolkit.py

```diff
@@ -0,0 +1,237 @@
+
+    return source
+
+
+class NotionToolkit(BaseToolkit):
+    r"""A toolkit for retrieving information from the user's notion pages.
+
+    Attributes:
+        notion_token (Optional[str], optional): The notion_token used to
+            interact with notion APIs.(default: :obj:`None`)
+        notion_client (module): The notion module for interacting with
+            the notion APIs.
+    """
+
+    def __init__(
+        self,
+        notion_token: Optional[str] = None,
+    ) -> None:
+        r"""Initializes the NotionToolkit."""
+        from notion_client import Client
+
+        self.notion_token = notion_token or os.environ.get("NOTION_TOKEN")
+
+        self.notion_client = Client(auth=self.notion_token)
+
+    def list_all_pages(self) -> List[dict]:
+        r"""Lists all pages in the Notion workspace.
+
+        Returns:
+            A list of page objects with title and id.
+        """
+        all_pages_info: List[dict] = []
+        cursor = None
+
+        while True:
+            response = cast(
+                dict,
+                self.notion_client.search(
+                    filter={"property": "object", "value": "page"},
+                    start_cursor=cursor,
+                ),
+            )
+            all_pages_info.extend(response["results"])
+
+            if not response["has_more"]:
+                break
+
+            cursor = response["next_cursor"]
+
+        final_list = [
+            {
+                "id": page["id"],
+                "title": next(
+                    (
+                        title["text"]["content"]
+                        for title in page["properties"]
+                        .get("title", {})
+                        .get("title", [])
+                        if title["type"] == "text"
+                    ),
+                    None,
+                ),
+            }
+            for page in all_pages_info
+        ]
+
+        return final_list
+
+    def get_notion_block_text_content(self, block_id) -> str:
+        """Retrieves the text content of a Notion block.
+
+        Args:
+            block_id (str): The ID of the Notion block to retrieve.
+
+        Returns:
+            str: The text content of a Notion block, containing all
+            the sub blocks.
+        """
+        blocks: List[dict] = []
+        cursor = None
+
+        while True:
+            response = cast(
+                dict,
+                self.notion_client.blocks.children.list(
+                    block_id=block_id, start_cursor=cursor
+                ),
+            )
+            blocks.extend(response["results"])
+
+            if not response["has_more"]:
+                break
+
+            cursor = response["next_cursor"]
+
+        block_text_content = " ".join(
+            [self.get_text_from_block(sub_block) for sub_block in blocks]
+        )
+
+        return block_text_content
+
+    def get_text_from_block(self, block) -> str:
```

Comment:
Can use some type hints for all these functions?

Reply:
Done.